nixos/modules/services/matrix/draupnir.nix

```
@@ -0,0 +1,306 @@
+        freeformType = format.type;
+        options = {
+          #region Readonly settings - these settings are not configurable
+          dataPath = lib.mkOption {
+            type = lib.types.str;
+            default = "/var/lib/draupnir";
+            readOnly = true;
+            description = ''
+              The path where Draupnir stores its data.
+
+              ::: {.note}
+                If you want to customize where this data is stored, use a bind mount.
+              :::
+            '';
+          };
+
+          pantalaimon = lib.mkOption {
+            readOnly = true;
+            description = ''
+              `pantalaimon` settings (enables E2E Encryption support).
+              This property is read-only, please configure `services.draupnir.pantalaimon` instead!
+            '';
+            default = {};
+            type = lib.types.submodule {
+              freeformType = format.type;
+              options = {
+                use = lib.mkOption {
+                  type = lib.types.bool;
+                  default = cfg.pantalaimon.enable;
+                  defaultText = "cfg.pantalaimon.enable";
+                  readOnly = true;
+                  description = ''
+                    Whether to use `pantalaimon` for E2E encryption. Enabled if `services.draupnir.pantalaimon.enable` is `true`.
+                  '';
+                };
+                username = lib.mkOption {
+                  type = lib.types.str;
+                  default = cfg.pantalaimon.username;
+                  defaultText = "cfg.pantalaimon.username";
+                  readOnly = true;
+                  description = ''
+                    Account name on the Matrix homeserver. Configured in `services.draupnir.pantalaimon.username`.
+                  '';
+                };
+              };
+            };
+          };
+          #endregion
+
+          #region Base settings
+          homeserverUrl = lib.mkOption {
+            type = lib.types.str;
+            defaultText = ''
+              URI to the pantalaimon instance, if configured. Else unset.
+            '';
+            description = ''
+              Base URL of the Matrix homeserver, that provides the Client-Server API.
+              If `services.draupnir.pantalaimon.enable` is `true`, this option will become read only. Configure `services.draupnir.pantalaimon.options.homeserver` instead in that case.
+              The listen address of `pantalaimon` will then become the `homeserverUrl` of `draupnir`.
+            '';
+          } // lib.optionalAttrs cfg.pantalaimon.enable {
+            default = "http://${config.services.pantalaimon-headless.instances."draupnir".listenAddress}:${toString config.services.pantalaimon-headless.instances."draupnir".listenPort}/";
+          };
+          #endregion
+
+          #region Common settings
+          autojoinOnlyIfManager = lib.mkOption {
+            type = lib.types.bool;
+            default = false;
```

```suggestion
            default = true;
```
[Upstream default](https://github.com/the-draupnir-project/Draupnir/blob/v2.0.0-beta.4/config/default.yaml#L46) is `true`. Is there any reason to change this?